### PR TITLE
Return server-side cql version for supported options

### PR DESCRIPTION
**Pull request by mpenick**
Fixes: https://github.com/datastax/cql-proxy/issues/70

### CODE REVIEW

File: proxycore/cluster.go [hunk 1]
```diff
@@ -33,33 +33,65 @@ const (
 	DefaultRefreshTimeout = 5 * time.Second
 )
 
+type Event interface {
```
Comment (mpenick) [inline proxycore/cluster.go:36]: Added a small, unrelated refactor here to not use `interface{}` for events.

File: proxy/proxy.go [hunk 2]
```diff
@@ -356,7 +356,10 @@ func (c *client) Receive(reader io.Reader) error {
 
 	switch msg := body.Message.(type) {
 	case *message.Options:
-		c.send(raw.Header, &message.Supported{Options: map[string][]string{"CQL_VERSION": {"3.0.0"}, "COMPRESSION": {}}})
+		c.send(raw.Header, &message.Supported{Options: map[string][]string{
+			"CQL_VERSION": {c.proxy.cluster.Info.CQLVersion},
```
Comment (mpenick) [inline proxy/proxy.go:360]: This is the fix.

Comment (joao-r-reis) [reply proxy/proxy.go:360]: Ah interesting, in ZDM we are just forwarding this request to one of the clusters (which one is determined by a config setting). The only request that we generate a response for is `system.peers` and `system.local` queries, everything else is forwarded.

File: proxycore/cluster.go (hunk 1 above)
Comment (joao-r-reis) [reply proxycore/cluster.go:36]: Nice

File: proxycore/cluster.go [hunk 3]
```diff
@@ -33,33 +33,65 @@ const (
...
+	isEvent() // Marker method for the event interface
+}
+
 type AddEvent struct {
 	Host *Host
 }
 
+func (a AddEvent) isEvent() {
+	panic("do not call")
```
Comment (joao-r-reis) [inline proxycore/cluster.go:45]: Why this `panic` on all of these methods?

**Review by joao-r-reis — APPROVED** [review]
Looks good, added a question about the `panic` calls.

File: proxycore/cluster.go (hunk 3 above)
Comment (mpenick) [reply proxycore/cluster.go:45]: This method should never be called because it's just a "dummy"/"marker" method because golang interfaces require at least one method.

Comment (joao-r-reis) [reply proxycore/cluster.go:45]: Ohh I see, thanks!
